util/topsql/topsql.go

```diff
@@ -40,10 +42,31 @@ var globalTopSQLReport reporter.TopSQLReporter
 
 // SetupTopSQL sets up the top-sql worker.
 func SetupTopSQL() {
-	rc := reporter.NewGRPCReportClient(plancodec.DecodeNormalizedPlan)
-	globalTopSQLReport = reporter.NewRemoteTopSQLReporter(rc)
+	globalTopSQLReport = reporter.NewRemoteTopSQLReporter(plancodec.DecodeNormalizedPlan)
+
 	tracecpu.GlobalSQLCPUProfiler.SetCollector(globalTopSQLReport)
 	tracecpu.GlobalSQLCPUProfiler.Run()
+
+	// register single target datasink to reporter
+	singleTargetDataSink := reporter.NewSingleTargetDataSink()
+	globalTopSQLReport.DataSinkRegHandle().Register(singleTargetDataSink)
+}
+
+// InstanceEnabled is used to check if TopSQL is enabled on the current instance.
+func InstanceEnabled() bool {
+	collector := tracecpu.GlobalSQLCPUProfiler.GetCollector()
+	if collector == nil {
+		return false
+	}
+	return !collector.IsPaused()
+}
```

Comment:
The name of this function is `InstanceEnabled`, which means that the entire TopSQL has been turned on. But the inner code is specific to the `tracecpu.Collector`. If the `Collector` of `StatementStats` is added later, will there be any problems if we continue to use only `tracecpu.Collector` to judge?

Reply:
How about `globalTopSQLReport` instead?

Reply:
> How about `globalTopSQLReport` instead?

Looks more reasonable~